src/devices/Camera/Camera/Settings/CommandOptionsBuilder.cs

```diff
@@ -0,0 +1,393 @@
+    }
+
+    /// <summary>
+    /// Retrieves all the command line options and values in a string array.
+    /// </summary>
+    /// <returns>An array of strings with all the command options accumulated in this instance.</returns>
+    public string[] GetArguments()
+    {
+        var args = _commands
+            .Select(c => $"{c.Option.Option}{(string.IsNullOrEmpty(c.Value) ? string.Empty : " ")}{c.Value}")
+            .ToArray();
+
+        return args;
+    }
+
+    /// <summary>
+    /// Adds the option specified in the argument.
+    /// </summary>
+    /// <param name="optionAndValue">The command option and its value to be added to the current instance.</param>
+    /// <returns>A reference to this instance.</returns>
+    public CommandOptionsBuilder With(CommandOptionAndValue optionAndValue)
+    {
+        _commands.Add(optionAndValue);
+        return this;
+    }
+
+    /// <summary>
+    /// Remove the specified option.
+    /// </summary>
+    /// <param name="command">The command to remove.</param>
+    /// <returns>A reference to this instance.</returns>
+    public CommandOptionsBuilder Remove(CommandOption command)
+    {
+        var optionsAndValues = _commands.Where(cv => cv.Option == command).ToArray();
+        foreach (var cv in optionsAndValues)
+        {
+            _commands.Remove(cv);
+        }
+
+        return this;
+    }
+
+    /// <summary>
+    /// Tells the app to output the text with all the installed cameras and their characteristics.
+    /// </summary>
+    /// <returns>A reference to this instance.</returns>
+    public CommandOptionsBuilder WithListCameras()
+    {
+        AddListCameras();
+        return this;
+    }
+
+    /// <summary>
+    /// Tells the app to use the camera with the specified index.
+    /// The indexes are obtained from the output when WithListCameras is used.
+    /// </summary>
+    /// <param name="index">The index of the camera to be used.</param>
+    /// <returns>A reference to this instance.</returns>
+    public CommandOptionsBuilder WithCamera(int index)
+    {
+        AddCamera(index);
+        return this;
+    }
+
+    /// <summary>
+    /// Tells the app to output the binary content towards the specified output.
+    /// </summary>
+    /// <param name="output">A valid option, URL or filename.</param>
+    /// <returns>A reference to this instance.</returns>
+    public CommandOptionsBuilder WithOutput(string output)
+    {
+        AddOutput(output);
+        return this;
+    }
+
+    /// <summary>
+    /// Adds the options to capture the stream for the given amount of milliseconds.
+    /// The value 0 (default) will capture indefinitely until the process gets stopped.
+    /// This option makes only sense for videos.
+    /// </summary>
+    /// <param name="ms">The length of the capture operation in milliseconds.</param>
+    /// <returns>A reference to this instance.</returns>
+    public CommandOptionsBuilder WithContinuousStreaming(int ms = 0)
```

Comment:
Nit: It would be really nice to convert these args to TimeSpan where possible but feel free to do it separately or ignore this since it's minor feedback

Reply:
If you don't mind I would discuss this in a later PR because I could not find a way to specify the default values as const
(TimeSpan.Zero is not const and `1 ms` as well)

Reply:
The following should work as default arguments: `new TimeSpan(0)` or `new TimeSpan(valueAs100nsTicks)`. As an alternative, you could use overloading instead of default arguments.

But I agree that's something that could be done separately.

Reply:
Ok, let's discuss it for later then :)